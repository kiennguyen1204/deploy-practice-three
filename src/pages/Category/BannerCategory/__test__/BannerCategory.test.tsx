import { render, fireEvent } from '@testing-library/react';

// components
import { BannerCategory } from '../index';

describe('BannerCategory component', () => {
  it('renders component correctly', () => {
    const { asFragment } = render(<BannerCategory />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('performs add to cart action', () => {
    const { getByText } = render(<BannerCategory />);
    const addToCartButton = getByText('Add to cart');
    fireEvent.click(addToCartButton);
  });
});
