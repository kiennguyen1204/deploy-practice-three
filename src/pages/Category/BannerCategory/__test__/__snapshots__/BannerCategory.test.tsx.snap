// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCategory component renders component correctly 1`] = `
<DocumentFragment>
  <div
    class="banner"
  >
    <article
      class="banner-bottom relative"
    >
      <div
        class="blur"
      />
      <div
        class="banner-item-wrapper"
      >
        <section
          class="flex-column wrapper-left"
        >
          <p
            class="cate"
          >
            CONCENTRATES
          </p>
          <h1
            class="title"
          >
            Mix And Match Shatter/Budder 28g (4 X 7G)
          </h1>
          <div
            class="align-center rating-n-review"
          >
            <span
              class="align-center rating"
            >
              <img
                alt="Star"
                src="IMAGE_MOCK"
              />
              <span
                class="star-text"
              >
                4.6/ 5
              </span>
            </span>
            <span
              class="sign"
            />
            <span
              class="align-center reviews"
            >
              135
              <span
                class="text-review"
              >
                Reviews
              </span>
            </span>
          </div>
          <div
            class="align-center volumes"
          >
            <span
              class="flex-container"
            >
              28g
            </span>
            <span
              class="flex-container"
            >
              1/2lb
            </span>
            <span
              class="flex-container"
            >
              1/4lb
            </span>
          </div>
          <div
            class="flex-space-between-container button-n-price"
          >
            <button
              class="btn btn-primary btn-color-default btn-sm   btn btn-add-cart-custom btn-color-text-primary"
              type="button"
            >
              <span>
                Add to cart
              </span>
            </button>
            <p
              class="price"
            >
              <span>
                $102.00
              </span>
              <span>
                $200.00
              </span>
            </p>
          </div>
        </section>
        <section
          class="wrapper-right"
        >
          <div
            class="background-blur-item relative"
          >
            <div
              class="splide__arrows"
            >
              <button
                class="splide__arrow splide__arrow--prev"
              >
                <img
                  alt="Previous"
                  src="IMAGE_MOCK"
                />
              </button>
              <button
                class="splide__arrow splide__arrow--next"
              >
                <img
                  alt="Next"
                  src="IMAGE_MOCK"
                />
              </button>
            </div>
            <img
              alt="banner-category"
              src="/src/assets/images/category/product_category.webp"
            />
          </div>
          <div
            class="flex-container dots"
          >
            <span
              class="dot active "
            />
            <span
              class="dot false "
            />
            <span
              class="dot false "
            />
            <span
              class="dot false "
            />
          </div>
        </section>
      </div>
    </article>
  </div>
</DocumentFragment>
`;
